File: Xamarin/MobileAzureDevDays.UITests/Pages/SentimentPage.cs
using System.Linq;

using Xamarin.UITest;

using MobileAzureDevDays.Shared.Constants;

using Query = System.Func<Xamarin.UITest.Queries.AppQuery, Xamarin.UITest.Queries.AppQuery>;

namespace MobileAzureDevDays.UITests
{
    public class SentimentPage : BasePage
    {
        readonly Query submitButton, enterTextBox, activityIndicator, emojiResultsLabel;

        public SentimentPage(IApp app) : base(app, PageTitles.SentimentPage)
        {
            submitButton = x => x.Marked(AutomationIdConstants.SubmitButton);
            enterTextBox = x => x.Marked(AutomationIdConstants.EnterTextBox);
            activityIndicator = x => x.Marked(AutomationIdConstants.ActivityIndicator);
            emojiResultsLabel = x => x.Marked(AutomationIdConstants.EmojiResultsLabel);
        }

        public string EmojiResultsText => App.Query(emojiResultsLabel)?.FirstOrDefault()?.Text;

        public void TapSubmitButton()
        {
            App.Tap(submitButton);
            App.Screenshot("Submit Button Tapped");
        }

        public void EnterText(string text)
        {
            App.EnterText(enterTextBox, text);
            App.Screenshot($"Entered Text: {text}");
        }

        public void WaitForActivityIndicator()
        {
            App.WaitForElement(activityIndicator);
            App.Screenshot("Activity Indicator Appeared");
        }

        public void WaitForNoActivityIndicator()
        {
            App.WaitForNoElement(activityIndicator);
            App.Screenshot("Activity Indicator Disappeared");
        }
    }
}

File: Swift/MobileAzureDevDays/MobileAzureDevDaysUITests/Pages/SentimentPage.swift
//
//  SentimentPage.swift
//  MobileAzureDevDaysUITests
//
//  Created by Brandon Minnick on 2/14/18.
//  Copyright © 2018 Colby Williams. All rights reserved.
//

import XCTest
import AppCenterXCUITestExtensions

class SentimentPage{
    
    var textView: XCUIElement
    var submitButton: XCUIElement
    var resultsLabel: XCUIElement
    var activityIndicator: XCUIElement
    
    init(app: XCUIApplication) {
        textView = app.textViews.element(boundBy: 0)
        submitButton = app.buttons["Submit"]
        resultsLabel = app.staticTexts.element(matching: .any, identifier: "EmojiLabel")
        activityIndicator = app.activityIndicators.element(matching: .any, identifier: "ActivityIndicator")
        
        ensureXCUIElementsExist()
    }
    
    public func tapSubmitButton() {
        submitButton.tap()
        ACTLabel.labelStep("Submit Button Tapped")
    }
    
    public func enterText(text: String){
        textView.tap()
        textView.typeText(text)
        ACTLabel.labelStep("Entered \(text)")
    }
    
    public func getResults() -> String{
        let labelText = resultsLabel.label
        return labelText
    }
    
    public func waitForActivityIndicator(timeout: TimeInterval){
        XCTAssertTrue((activityIndicator.waitForExistence(timeout: timeout)))
        ACTLabel.labelStep("Activity Indicator Appeared")
    }
    
    public func waitForNoActivityIndicator(timeout: TimeInterval){
        XCTAssertFalse((activityIndicator.waitForExistence(timeout: timeout)))
        ACTLabel.labelStep("Activity Indicator Disappeared")
    }
    
    func ensureXCUIElementsExist() {
        XCTAssertTrue(textView.exists)
        XCTAssertTrue(submitButton.exists)
        XCTAssertTrue(resultsLabel.exists)
    }
}
